EXAMPLES.md

```diff
@@ -231,3 +231,33 @@ app.use(auth({
   }
 }));
 ```
+
+## 8. Custom state handling
+
+If your application needs to keep track of the request state before redirecting to log in, you can use the built-in state handling. By default, this library stores the post-callback redirect URL in a state object (along with a generated nonce) that is converted to a string, base64 encoded, and verified during callback (see [our documentation](https://auth0.com/docs/protocols/oauth2/oauth-state) for general information about this parameter). This state object can be added to and used during callback.
+
+You can define a `getLoginState` configuration key set to a function that takes an Express `RequestHandler` and an options object and returns a URL-safe string representation:
+
+```js
+const crypto = require('crypto');
+
+app.use(auth({
+  getLoginState: function (req, options) {
+    const state = {
+      // Property used by the library for redirecting after logging in.
+      returnTo: '/custom-return-path',
+      // Required to make sure the state parameter can't be replayed.
+      nonce: crypto.randomBytes(32).toString('hex');
+      // Additional properties as needed.
+      customProperty: req.someProperty,
+    };
+    // This value will be sent in a URL parameter so it should be transfer-safe.
+    return req.openid.encodeState(state);
+  },
+  handleCallback: function (req, res, next) {
+    const decodedState = req.openid.decodeState(eq.openidState);
```

Comment:
If we move the encode operation into the library (see above) we'd also probably want to have decoded it by now rather than put the onus on the developer for that.